features/device_key/TESTS/device_key/functionality/main.cpp

```diff
@@ -0,0 +1,492 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "DeviceKey.h"
+#include "utest/utest.h"
+#include "unity/unity.h"
+#include "greentea-client/test_env.h"
+#include "nvstore.h"
+
+#if !NVSTORE_ENABLED
+#error [NOT_SUPPORTED] NVSTORE needs to be enabled for this test
+#endif
+
+using namespace utest::v1;
+
+#define MSG_VALUE_DUMMY "0"
+#define MSG_VALUE_LEN 32
+#define MSG_KEY_LEN 32
+
+#define MSG_KEY_DEVICE_READY "ready"
+#define MSG_KEY_DEVICE_FINISH "finish"
+#define MSG_KEY_DEVICE_TEST_STEP1 "check_consistency_step1"
+#define MSG_KEY_DEVICE_TEST_STEP2 "check_consistency_step2"
+#define MSG_KEY_DEVICE_TEST_STEP3 "check_consistency_step3"
+#define MSG_KEY_DEVICE_TEST_STEP4 "check_consistency_step4"
+#define MSG_KEY_DEVICE_TEST_SUITE_ENDED "Test suite ended"
+
+void generate_derived_key_consistency_16_byte_key_reset_test(char *key);
+void generate_derived_key_consistency_32_byte_key_reset_test(char *key);
+
+/*
+ * Injection of a dummy key when there is no TRNG
+ */
+int inject_dummy_rot_key()
+{
+#if !defined(DEVICE_TRNG)
+    uint32_t key[DEVICE_KEY_16BYTE / sizeof(uint32_t)];
+
+    memset(key, 0, DEVICE_KEY_16BYTE);
```

Comment:
This `memset` call is redundant because of the subsequent `memcpy`.

Reply:
@hanno-arm 
I have removed the memset.